File: python/matrix/matrix.py
"""
matrix parsing from string
"""
class Matrix:
    """
    Matrix class to parse string
    """


    def __init__(self, matrix_string):
        """
        split string and parse elements to int
        """
        self.rows = [[int(i) for i in row.split()] for row in matrix_string.split("\n")]

    def row(self, index):
        """
        get single row
        """
        return self.rows[index-1]

    def column(self, index):
        """
        build a column
        """
        return [row[index-1] for row in self.rows]

File: ruby/matrix/matrix.rb
class Matrix
  attr_reader :rows, :columns

  def initialize(rows)
    @rows = []
    @columns = []
    rows.split(/\n/).each do |row|
      @rows << row.split.map(&:to_i)
      row.split.each_with_index do |cell, index|
        @columns[index] ||= []
        @columns[index] << cell.to_i
      end
    end
  end
end
